server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -1002,10 +1020,54 @@ private TemplateVariables buildTemplateVariables(CreateFlowInstanceReq flowInsta
+        variables.setAttribute(Variable.PROJECT_OWNER_IDS, JsonUtils.toJson(projectOwnerIds));
+        List<String> projectOwnerAccounts =
+                projectOwners.stream().map(User::getAccountName).collect(Collectors.toList());
+        variables.setAttribute(Variable.PROJECT_OWNER_ACCOUNTS, JsonUtils.toJson(projectOwnerAccounts));
+        List<String> projectOwnerNames = projectOwners.stream().map(User::getName).collect(Collectors.toList());
+        variables.setAttribute(Variable.PROJECT_OWNER_NAMES, JsonUtils.toJson(projectOwnerNames));
+        // set database related variables
+        if (Objects.nonNull(flowInstanceReq.getDatabaseId())) {
+            Database database = databaseService.detail(flowInstanceReq.getDatabaseId());
+            variables.setAttribute(Variable.DATABASE_NAME, database.getName());
+            if (Objects.nonNull(database.getEnvironment())) {
+                String environmentNameKey = database.getEnvironment().getName();
+                if (StringUtils.isTranslatable(environmentNameKey)) {
+                    String environmentName = I18n.translate(StringUtils.getTranslatableKey(environmentNameKey), null,
+                            LocaleContextHolder.getLocale());
+                    variables.setAttribute(Variable.ENVIRONMENT_NAME, environmentName);
+                }
+            }
+            List<User> databaseOwners = new ArrayList<>();
+            List<UserResourceRole> userResourceRoles =
+                    resourceRoleService.listByResourceTypeAndId(ResourceType.ODC_DATABASE, database.getId());
+            if (CollectionUtils.isNotEmpty(userResourceRoles)) {
+                Set<Long> userIds =
+                        userResourceRoles.stream().map(UserResourceRole::getUserId).collect(Collectors.toSet());
+                databaseOwners = userService.batchNullSafeGet(userIds);
+            } else {
+                databaseOwners = projectOwners;
+            }
+            List<Long> ownerIds = databaseOwners.stream().map(User::getId).collect(Collectors.toList());
+            variables.setAttribute(Variable.DATABASE_OWNERS_IDS, JsonUtils.toJson(ownerIds));
+            List<String> ownerAccounts = databaseOwners.stream().map(User::getAccountName).collect(Collectors.toList());
+            variables.setAttribute(Variable.DATABASE_OWNERS_ACCOUNTS, JsonUtils.toJson(ownerAccounts));
+            List<String> ownerNames = databaseOwners.stream().map(User::getName).collect(Collectors.toList());
+            variables.setAttribute(Variable.DATABASE_OWNERS_NAMES, JsonUtils.toJson(ownerNames));
+        }
         // set SQL content if task type is DatabaseChange
         if (taskType == TaskType.ASYNC) {
             DatabaseChangeParameters params = (DatabaseChangeParameters) flowInstanceReq.getParameters();
-            String sqlContent = params.getSqlContent();
+            String sqlContent = JsonUtils.toJson(params.getSqlContent());
```

Comment:
Why do we need to transform sql to json

Reply:
If the SQL content has character such as `"`, and we use the template variable as:
```json
{
    "sql": "${sql.content}"
}
```
Then the json content is invalid.